test/WebJobs.Script.Tests/ScriptHostTests.cs

```diff
@@ -1488,6 +1488,53 @@ public void IsFunction_ReturnsExpectedResult()
             Assert.False(host.IsFunction(null));
         }
 
+        [Fact]
+        public void Initialize_LogsWarningForExplicitlySetHostId()
+        {
+            TestLoggerProvider loggerProvider = null;
+            var loggerFactoryHookMock = new Mock<ILoggerProviderFactory>(MockBehavior.Strict);
+            loggerFactoryHookMock
+                .Setup(m => m.CreateLoggerProviders(It.IsAny<ScriptHostConfiguration>(), It.IsAny<ScriptSettingsManager>(), It.IsAny<Func<bool>>(), It.IsAny<Func<bool>>()))
+                .Returns<ScriptHostConfiguration, ScriptSettingsManager, Func<bool>, Func<bool>>((scriptConfig, settingsManager, fileLoggingEnabled, isPrimary) =>
+                {
+                    loggerProvider = new TestLoggerProvider(scriptConfig.LogFilter.Filter);
+                    return new[] { loggerProvider };
+                });
+
+            string rootPath = Path.Combine(Environment.CurrentDirectory, "ScriptHostTests");
```

Comment:
Is the ScriptHostTests directory used by any other tests? I just want to make sure we don't muck with any other tests that expects an empty host.json at that location. To be safe, you may just want to use the name of this test as the directory name -- to make sure it only ever affects this one.